### DISCUSSION TITLE
Unable to get selected rows data

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#11965] [Q&A] [answered]
Hello,

I want to fetch selected rows data from my data grid. But i am getting the following error.
Note: I am using IGX version 10.x 

![image](https://user-images.githubusercontent.com/101707048/184292356-9b4f3173-1f91-4022-bb39-29aa69406029.png)

Here is my Html tag

![image](https://user-images.githubusercontent.com/101707048/184292450-b41a7ceb-edd8-4ea7-96e6-4fba65b8653f.png)

Here is my code in ts file 

![image](https://user-images.githubusercontent.com/101707048/184292572-6e49c57a-7ecc-4f30-945f-fb89e0c58c6b.png)

I am calling the below function on click 
![image](https://user-images.githubusercontent.com/101707048/184292646-ca50a73a-7740-4a55-ab30-282d7fc1d6a2.png)

But it is giving me above error
Please respond with a solution to fetch selected row data in current version

**Comment by georgianastasov** [comment, marked as answer]
Hello @kevalgw,

The way you try to set the selected rows in the igx-grid and then retrieve them is the correct way to do it, but for version 14.x of Ignite UI Angular, as there have been several changes between version 10 and 14.

First, the [selectedRows](https://www.infragistics.com/products/ignite-ui-angular/docs/14.0.14/typescript/classes/IgxGridComponent.html#selectedRows) property on the igx-grid is not available in version 10.x. It has been introduced in the newer supported versions of Ignite UI Angular. What I can suggest is to use the [selectRows](https://www.infragistics.com/products/ignite-ui-angular/docs/10.0.12/typescript/classes/igxgridcomponent.html#selectrows)() method and select the rows you want in the ngAfterViewInit hook.

Second, retrieving the selected rows in version 10.x can be done with the [selectedRows](https://www.infragistics.com/products/ignite-ui-angular/docs/10.0.12/typescript/classes/igxgridcomponent.html#selectedrows)() method. In later supported versions, this is done with the [selectedRows](https://www.infragistics.com/products/ignite-ui-angular/docs/14.0.14/typescript/classes/IgxGridComponent.html#selectedRows) accessor.

Having in mind that igniteui-angular version 10.x is considered retired and is no longer eligible for Developer Support Services, to achieve your requirement and take advantage of all introduced features and enhancements, I would suggest upgrading to the latest version i.e. 14.

In addition, I have prepared a small sample demonstrating my suggestion and it could be found [here](https://stackblitz.com/edit/angular-5szwrj?file=src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html).

**Reply by kevalgw** [reply]
Hi @georgianastasov 

Thanks for the reply ,
